pkg/runner/generator.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package runner
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/opensbom-generator/parsers/meta"
+	"github.com/opensbom-generator/parsers/plugin"
+
+	"github.com/spdx/spdx-sbom-generator/pkg/runner/options"
+)
+
+type Generator struct {
+	Options        options.Options
+	implementation GeneratorImplementation
+	docHandler     DocumentFormatHandler
+}
+
+type DocumentFormatHandler interface {
+	CreateDocument(*options.Options) (interface{}, error)
+	AddDocumentPackages(*options.Options, interface{}, []meta.Package) error
```

Comment:
Why not combine these together? I remember you had a reason, but I can't recall what it was.

Reply:
I split them in preparation for a more complete generation process. If we add more steps, for example splitting the licensing bits or registering files, then it is easier to move the stages around. 

Since it is is internal, we can combine them now and split them later if we need it.

Reply:
Based on conversations we've been having in the SBOM community (I'm thinking about our current issue with CreationInfo) it seems to me we will probably want to generate a document with its elements all in one go. In which case, maybe the underlying functionality (make each element one by one and assemble them into a document) maybe makes more sense?